tutorials/fundamentals/clear-array.md

```diff
@@ -0,0 +1,36 @@
+There are a few ways to clear an array in JavaScript.
+
+## array.length = 0
+
+The simplest solution is to assign the array's length to the value of 0.
+The length property is readable and writeable which makes this possible.
+
+```javascript
+let array = [1,2,3,4,5];
+array.length = 0;
+array; // []
+```
+
+## array.splice(0, array.length)
+
+A more verbose approach is to use the `splice()` method.
+This function will return a copy of the array, so if you wanted to do a reassignment this would be a good use case.
+
+```javascript
+let array = [1,2,3,4,5];
+let anotherArray = array.splice(0, array.length);
+anotherArray; // [1,2,3,4,5]
+array; // []
+```
+
+## array = []
+
+Reassinging the array to an empty array has some repercussions.
```

Comment:
Replace this with "If you want to avoid mutating arrays in place for [immutability](https://reactjs.org/docs/update.html), you can overwrite the array with an empty array `[]`."

And example should use an object:

```javascript
let obj = { array: [1, 2, 3, 4, 5] };

obj = { ...obj, array: [] };
obj.array.length; // 0
```

Reply:
I was trying to show in this section more so an incorrect way of emptying an array as it is probably the first thing newbies try when they want to reset their array. Doing `array.length = 0` or `array.splice(0, array.length)` aren't intuitive but they would have enough experience that it might come to mind to just do a reassignment like how if they wanted to empty a string they would have done `string = ""`. I will add your requested change but I would like to keep that section as it would provide answer for those wondering why doing `array = []` is causing them problems.